extensions/csv/src/main/java/io/deephaven/csv/CsvTools.java

```diff
@@ -913,35 +885,321 @@ private static void writeCsvContentsSeq(
+                if (value == NULL_CHAR) {
+                    writeNull(writer);
+                    return;
+                }
+                writer.write(separatorCsvEscape(String.valueOf(value), separator));
+            }
+        }
+
+        private final class ByteColumnFormatter extends ColumnFormatter<ByteChunk<? extends Any>> {
+
+            private ByteColumnFormatter(@NotNull final ChunkSource<? extends Any> source) {
+                super(source);
+            }
+
+            @Override
+            void write(@NotNull final Writer writer, final int offset) throws IOException {
+                final byte value = values.get(offset);
+                if (value == NULL_BYTE) {
+                    writeNull(writer);
+                    return;
+                }
+                writer.write(separatorCsvEscape(Byte.toString(value), separator));
+            }
+        }
+
+        private final class ShortColumnFormatter extends ColumnFormatter<ShortChunk<? extends Any>> {
+
+            private ShortColumnFormatter(@NotNull final ChunkSource<? extends Any> source) {
+                super(source);
+            }
+
+            @Override
+            void write(@NotNull final Writer writer, final int offset) throws IOException {
+                final short value = values.get(offset);
+                if (value == NULL_SHORT) {
+                    writeNull(writer);
+                    return;
+                }
+                writer.write(separatorCsvEscape(Short.toString(value), separator));
+            }
+        }
+
+        private final class IntColumnFormatter extends ColumnFormatter<IntChunk<? extends Any>> {
+
+            private IntColumnFormatter(@NotNull final ChunkSource<? extends Any> source) {
+                super(source);
+            }
+
+            @Override
+            void write(@NotNull final Writer writer, final int offset) throws IOException {
+                final int value = values.get(offset);
+                if (value == NULL_INT) {
+                    writeNull(writer);
+                    return;
                 }
-                if (progress != null) {
-                    progress.accept(ri, size);
+                writer.write(separatorCsvEscape(Integer.toString(value), separator));
+            }
+        }
+
+        private final class LongColumnFormatter extends ColumnFormatter<LongChunk<? extends Any>> {
+
+            private LongColumnFormatter(@NotNull final ChunkSource<? extends Any> source) {
+                super(source);
+            }
+
+            @Override
+            void write(@NotNull final Writer writer, final int offset) throws IOException {
+                final long value = values.get(offset);
+                if (value == NULL_LONG) {
+                    writeNull(writer);
+                    return;
+                }
+                writer.write(separatorCsvEscape(Long.toString(value), separator));
```

Comment:
Do we ever expect any of the primitives to contain characters that need escaping? I guess somebody could use something like "0" as a separator (I don't know if that's realistic, or if other things would break in that case...)?

Reply:
We allow crazy separators, and unit test same. The fix to header escaping came about because `Z` as a separator was causing issues with a column header named `ZonedDateTimes`.